p2panda-rs/src/storage_provider/traits/entry_store.rs

```diff
@@ -18,29 +18,137 @@ use crate::storage_provider::traits::AsStorageEntry;
 #[async_trait]
 pub trait EntryStore<StorageEntry: AsStorageEntry> {
     /// Insert an entry into storage.
+    ///
+    /// Returns `true` if insertion was succesful, returns `false` if no error occured but an
+    /// unexpected number of insertions happened. Errors if a fatal storage error occured.
```

Comment:
Which scenario would cause an unexpected number of insertions without being a fatal error? Not saying the API should be different, I am just not sure what that comment means.

Reply:
My preferred approach would probably be be to still check, but instead of returning `true` or `false` we just error if an unexpected number of insertions happened.